File: spring-web-java/src/main/java/spring/web/java/presentation/exception/OrderCancelNotAllowedException.java
package spring.web.java.presentation.exception;

import org.springframework.http.HttpStatus;

public class OrderCancelNotAllowedException extends BusinessException {

	public OrderCancelNotAllowedException(Long orderId) {
		super("주문(ID: %d)은 현재 상태에서는 취소할 수 없습니다.".formatted(orderId), HttpStatus.UNPROCESSABLE_ENTITY);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/exception/OrderCancelNotAllowedException.kt
package spring.web.kotlin.presentation.exception

import org.springframework.http.HttpStatus

class OrderCancelNotAllowedException(orderId: Long) :
    BusinessException("주문(ID: $orderId)은 현재 상태에서는 취소할 수 없습니다.", HttpStatus.UNPROCESSABLE_ENTITY)